ca/config.go

```diff
@@ -75,6 +79,33 @@ func (s *SecurityConfig) RootCA() *RootCA {
 	return s.rootCA
 }
 
+// ExternalCA returns a client for external CA endpoints which may be used to
+// make certificate signing requests if no local RootCA signer is available.
+func (s *SecurityConfig) ExternalCA() *ExternalCA {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// Make a new Client TLS config and clear the ServerName field.
+	tlsConfig := &tls.Config{
+		Certificates: s.ClientTLSCreds.config.Certificates,
+		RootCAs:      s.ClientTLSCreds.config.RootCAs,
```

Comment:
I don't think it's safe to access these items inside `ClientTLSCreds` without holding the mutex inside `ClientTLSCreds`. I think it would be a good idea to add accessor functions for them.